### PR TITLE
fix: don't assume document.body covers viewport

### LINKED ISSUE #1367: useInteractOutside should not assume document.body covers viewport
# 🐛 Bug Report 

When [the `useOverlayTrigger` example](https://react-spectrum.adobe.com/react-aria/useOverlayTrigger.html#example) from the docs is viewed standalone (see [this Codesandbox](https://codesandbox.io/s/overlay-trigger-dd9p1?file=/src/App.tsx)), the overlay can only be dismissed by clicking the small area covered by `document.body` (yellow background in the Codesandbox).

## 🤔 Expected Behavior

Clicking anywhere outside the overlay should close it.

## 😯 Current Behavior

Overlay not closed.

## 💁 Possible Solution

I think this happens because `useInteractOutside` has [this check](https://github.com/adobe/react-spectrum/blob/1d9905e7c8308cfbb2a1e03bfc22b2dce89cbb12/packages/%40react-aria/interactions/src/useInteractOutside.ts#L114) (added in #715) which assumes `document.body` covers the viewport. If this is not the case, `isPointerDown` is not set to `true` [here](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/interactions/src/useInteractOutside.ts#L44-L46), and subsequently `onInteractOutside` does not get called [here](https://github.com/adobe/react-spectrum/blob/1d9905e7c8308cfbb2a1e03bfc22b2dce89cbb12/packages/%40react-aria/interactions/src/useInteractOutside.ts#L86-L89).

When changing `document.body` to `document.documentElement` everything seems to work fine. If this makes sense, i'll send a pr.

## 💻 Code Sample

see https://codesandbox.io/s/overlay-trigger-dd9p1?file=/src/App.tsx

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | latest
| Browser          | firefox 83
| Operating System | ubuntu 20.04

### PR DESCRIPTION
**Pull request by stefanprobst**
Closes #1367

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Review by snowystinger — COMMENTED** [review]
Thanks for figuring that out! If it's not too much trouble, would you be willing to put your example into a story so we can verify this in the future?

**Review by snowystinger — APPROVED** [review]

File: packages/@react-aria/interactions/stories/styles.css
Comment (LFDanLu) [inline packages/@react-aria/interactions/stories/styles.css:3]: Looks like these styles are being applied to the body of all stories, not just the useInteractOutside story. Lemme see if I can figure out why

Comment (snowystinger) [reply packages/@react-aria/interactions/stories/styles.css:3]: huh... that's annoying, thanks for catching that
we can probably make a component that applies these styles on mount and removes them on unmount, then just decorate this story with that

**Review by snowystinger — APPROVED** [review]
LGTM, I'll get the team to do a second review

**Review by ktabors — APPROVED** [review]
LGTM

**Review by devongovett — APPROVED** [review]

**Pull request merged**
